app/controllers/users_controller.rb

```diff
@@ -27,9 +27,31 @@ def update_profile_picture
     end
   end
 
+  def edit_password
+    @user = current_user
+  end
+  
+  def update_password
+    @user = current_user
+  
+    if @user.valid_password?(user_params[:current_password])
+      if @user.update_with_password(user_params)
```

Comment:
How we are using update_with_password method? i don't see its been defined in user mode?